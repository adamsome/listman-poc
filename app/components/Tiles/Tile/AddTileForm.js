import React from 'react';
import onClickOutside from 'react-onclickoutside'

import './AddTileForm.scss'

export class AddTileForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      addingName: "",
    };
  }

  handleKeyDown(e) {
    const { add, close } = this.props
    if (e.which === 13) {
      // Enter key pressed
      if (e.target.value.trim().length !== 0) {
        add(e.target.value.trim());
      }
      this.setState({ addingName: "" });
      close()

    } else if (e.which === 27) {
      // Esc key pressed
      this.setState({ addingName: "" });
      close()
    }
  }

  handleClickOutside(e) {
    this.props.close()
  }

  render() {
    return (
      <div>
        <p>
          <input
            className="tile__add-form-name input is-large"
            type="text"
            placeholder="Name..."
            value={this.state.addingName}
            autoFocus="true"
            onChange={(e) => this.setState({ addingName: e.target.value })}
            onKeyDown={::this.handleKeyDown}
          />
        </p>
        <p className="tile__add-form-instructions control">
          Enter to add, Esc to cancel
        </p>
      </div>
    )
  }
}

export default onClickOutside(AddTileForm)
