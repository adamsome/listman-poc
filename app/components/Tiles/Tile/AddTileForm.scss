/* Form input is inside of the Bulma .box */
.tile__add-form-name {
    font-weight: 300;
    height: 1.285em;
    padding-left: 0;
    padding-right: 0;
    border: none;
    box-shadow: none;
    margin-bottom: 0.25rem;
}

/* Light colored instructions */
.tile__add-form-instructions {
    color: hsl(0, 0%, 71%);
    margin-bottom: 0;
}
