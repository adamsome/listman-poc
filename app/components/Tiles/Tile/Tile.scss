.tile {
  padding-bottom: 0.5rem;
}
.tile__title-wrapper {
  display: block;
  min-height: 60px;
}

.tile__title {
  display: inline-block;
  overflow: hidden;
  max-height: 48px;
}

