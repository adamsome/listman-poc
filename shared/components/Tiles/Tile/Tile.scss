@import '../../../style/theme';

.box-tile {
  display: block;
  background-color: $white;
  padding: 1.25rem 1.25rem 0.5rem 1.25rem;
  box-shadow: 0 2px 3px rgba($black, 0.1), 0 0 0 1px rgba(10, 10, 10, 0.1);
  border-radius: 5px;
}

a.box-tile:hover, a.tile:focus {
  box-shadow: 0 2px 3px rgba($black, 0.1), 0 0 0 1px $link;
}

a.box-tile:active {
  box-shadow: inset 0 1px 2px rgba($black, 0.2), 0 0 0 1px $link;
}

/* Simple light grey BG while loading */
.box-tile--loading {
  background-color: $light;
  box-shadow: none;
}

/* Hide any lines over 2 */
.box-tile__title-wrapper {
  display: block;
  min-height: 4.25rem;
}

.box-tile__title {
  display: inline-block;
  overflow: hidden;
  max-height: 3.5rem;
  word-break: break-word;
  color: $grey-darker;
  font-size: $size-4;
  font-weight: $weight-light;
  line-height: 1.125;
}

