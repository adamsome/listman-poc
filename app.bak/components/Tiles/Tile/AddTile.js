import React from 'react';

import AddTileForm from './AddTileForm'
import './AddTile.scss'

export default class AddTile extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isFormOpen: false,
    };
  }

  render() {
    const { isLoading, isAdding, add } = this.props

    let className = "box tile__add"
    if (isLoading) {
      className += " tile--loading"
    } else if (this.state.isFormOpen) {
      className += " tile__add--form-open"
    } else {
      className += " level"
    }

    if (isAdding) {
      className += " tile__add--adding is-disabled"
    }

    let titleClassName = "tile__add__title title"
    if (isAdding) {
      titleClassName += " tile__add__title--adding"
    }

    return (
      <a
        className={className}
        onClick={() => this.setState({ isFormOpen: true })}
      >
        <span className="tile__add__title-wrapper level-item has-text-centered">
          {
            (!this.state.isFormOpen)
              ? <span className={titleClassName}>
                  {(isAdding) ? 'Adding...' : (isLoading) ? ' ' : '+'}
                </span>
              : <AddTileForm
                  add={add}
                  close={() => this.setState({ isFormOpen: false })}
                />
          }
        </span>
      </a>
    )
  }
}

