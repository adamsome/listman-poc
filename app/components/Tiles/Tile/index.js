import React from 'react';

import './Tile.scss'

const Tile = (props) => {
  return (
    <a className="tile box">
      <span className="tile__title-wrapper">
        <span className="tile__title title is-4">
          {props.name}
        </span>
      </span>
    </a>
  )
}

export default Tile
