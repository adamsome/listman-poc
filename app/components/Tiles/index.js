import React from 'react'

import Tile from './Tile'

const renderTilesLoading = () => {
  return <div>Loading</div>
}

const renderTiles = (entities, getEntityName = (entity => entity.name)) => {
  if (!entities || !entities.length) {
    return <div>No Items!</div>
  }
  return entities.map((entity, index) => {
    return (
      <div key={index} className="column is-one-quarter">
        <Tile
          key={index}
          name={getEntityName(entity)}
        />
      </div>
    )
  })
}

// TODO: Add reactProps
const Tiles = ({ entities, isLoading }) => (
    <div className="columns is-multiline">
      {(isLoading) ? renderTilesLoading() : renderTiles(entities)}
    </div>
)

export default Tiles

