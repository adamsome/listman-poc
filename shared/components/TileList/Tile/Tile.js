import React from 'react';
import classNames from 'classnames'

import './Tile.scss'

const Tile = ({ name, isLoading }) => {
  const className = classNames({
    "box-tile": true,
    "box-tile--loading": isLoading,
  })
  return (
    <a className={className}>
      <span className="box-tile__title-wrapper">
        <span className="box-tile__title">
          {name || ' '}
        </span>
      </span>
    </a>
  )
}

export default Tile
