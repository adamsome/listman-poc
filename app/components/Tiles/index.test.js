import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Tiles from './index'
import Tile from './Tile'

it('renders null based on initial state (empty "list" array)', () => {
  const wrapper = shallow(<Tiles entities={[]}/>)
  expect(wrapper.contains(<div>No Items!</div>)).toBe(true)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders <div> w/ correct children of type <Tile>', () => {
  const name = 'Test One'
  const wrapper = shallow(<Tiles entities={[ { name } ]}/>)
  expect(wrapper.find(Tile)).toHaveLength(1)

  const child = wrapper.find(Tile).first()
  const childProps = child.props()
  expect(childProps.name).toEqual(name)
  expect(toJson(wrapper)).toMatchSnapshot()
})

