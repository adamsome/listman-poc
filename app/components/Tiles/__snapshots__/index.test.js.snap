exports[`test renders <div> w/ correct children of type <Tile> 1`] = `
<div
  className="columns is-multiline">
  <div
    className="column is-one-quarter">
    <Tile
      name="Test One" />
  </div>
</div>
`;

exports[`test renders null based on initial state (empty "list" array) 1`] = `
<div
  className="columns is-multiline">
  <div>
    No Items!
  </div>
</div>
`;
